File: backend/POC.AURA.Api/Models/PrintModels.cs
namespace POC.AURA.Api.Models;

/// <summary>Inbound request from an Angular UI client to queue a print job.</summary>
public record PrintJobRequest(string DocumentName, string Content, int Copies = 1);

/// <summary>
/// Immutable snapshot of a print job routed to the SmartHub processor.
/// Serialised as a SignalR payload for <c>ExecutePrintJob</c> and <c>PrintJobQueued</c>.
/// </summary>
public record PrintJob(
    string Id,
    string TenantId,
    string DocumentName,
    string Content,
    int    Copies,
    string RequestorConnectionId,
    DateTime CreatedAt
);

/// <summary>
/// Result emitted to Angular via <c>PrintJobComplete</c> / <c>PrintJobStatusUpdate</c>
/// after the SmartHub reports completion.
/// </summary>
public record PrintJobResult(
    string   JobId,
    string   TenantId,
    string   RequestorConnectionId,
    bool     Success,
    string   Message,
    DateTime CompletedAt
);

/// <summary>
/// Request body POSTed by the SmartHub to <c>/api/print/complete</c>.
/// </summary>
public record CompleteJobRequest(string JobId, bool Success, string Message);

File: frontend/src/app/core/models/print.models.ts
/**
 * Represents a print job as received from AuraHub (`PrintJobQueued` / `ExecutePrintJob`).
 */
export interface PrintJob {
  id: string;
  tenantId: string;
  documentName: string;
  content: string;
  copies: number;
  requestorConnectionId: string;
  createdAt: string;
}

/**
 * Result emitted by AuraHub after the SmartHub reports job completion
 * (`PrintJobComplete` / `PrintJobStatusUpdate`).
 */
export interface PrintJobResult {
  jobId: string;
  tenantId: string;
  requestorConnectionId: string;
  success: boolean;
  message: string;
  completedAt: string;
}
